.github/workflows/cd.yml

```diff
@@ -12,143 +15,183 @@ on:
+  #     - name: Test docker install
+  #       run: |
+  #         docker build  -f Dockerfiles/pip.Dockerfile -t jinaai/jina:test-pip .
+  #         docker build --build-arg PIP_TAG="[devel]" -f Dockerfiles/pip.Dockerfile -t jinaai/jina:test-pip .
+  #     - name: Test with pytest
+  #       run: |
+  #         # make sure docker exist
+  #         docker ps
+  #         jina check
+  #         export JINA_LOG_VERBOSITY="ERROR"
+  #         pip install .[test]
+  #         pytest --force-flaky --min-passes 1 --cov=jina --cov-report=xml -n 1 --timeout=120 -v tests/unit
+  #         mv coverage.xml unit-test-coverage.xml
+  #       timeout-minutes: 30
+  #       env:
+  #         JINAHUB_USERNAME: ${{ secrets.JINAHUB_USERNAME }}
+  #         JINAHUB_PASSWORD: ${{ secrets.JINAHUB_PASSWORD }}
+  #     - name: Upload coverage result from unit-test
+  #       if: ${{ matrix.python-version }} == 3.7 && ${{ matrix.os }} == ubuntu-latest
+  #       uses: actions/upload-artifact@v1
+  #       with:
+  #         name: unit-test-codecoverage
+  #         path: ./unit-test-coverage.xml
+
+  # integration-test:
+  #   needs: update-docker
+  #   runs-on: ${{ matrix.os }}
+  #   strategy:
+  #     matrix:
+  #       os: [ubuntu-latest] #[macos-latest, ubuntu-latest]
+  #       python-version: [3.7,]
 
-  integration-test:
-    needs: update-docker
+  #   steps:
+  #     - uses: actions/checkout@v2
+  #       with:
+  #         submodules: true
+  #     - name: Set up Python ${{ matrix.python-version }}
+  #       uses: actions/setup-python@v1
+  #       with:
+  #         python-version: ${{ matrix.python-version }}
+  #     - name: Install dependencies
+  #       run: |
+  #         python -m pip install --upgrade pip
+  #         pip install .[match-py-ver] --no-cache-dir
+  #     - name: Test with pytest
+  #       run: |
+  #         # make sure docker exist
+  #         docker ps
+  #         jina check
+  #         docker build  -f Dockerfiles/pip.Dockerfile -t jinaai/jina:test-pip .
+  #         export JINA_LOG_VERBOSITY="ERROR"
+  #         pip install .[test]
+  #         pytest --force-flaky --min-passes 1 --cov=jina --cov-report=xml -n 1 --timeout=120 --ignore-glob='tests/integration/hub_usage/dummyhub*' -v tests/integration
+  #         mv coverage.xml integration-test-coverage.xml
+  #       timeout-minutes: 30
+  #       env:
+  #         JINAHUB_USERNAME: ${{ secrets.JINAHUB_USERNAME }}
+  #         JINAHUB_PASSWORD: ${{ secrets.JINAHUB_PASSWORD }}
+  #     - name: Upload coverage result from integration-test
+  #       if: ${{ matrix.python-version }} == 3.7 && ${{ matrix.os }} == ubuntu-latest
+  #       uses: actions/upload-artifact@v1
+  #       with:
+  #         name: integration-test-codecoverage
+  #         path: ./integration-test-coverage.xml
+
+  examples-test:
```

Comment:
I would suggest to put first in the `ci.yml` so that at every commit or push forced it will be executed and u can see if the action works, then later it can be moved to the `cd.yml`. @maximilianwerk  was doing this for his black formating PR

Reply:
Yes same thoughts!

Reply:
Check the added `examples-test` job and let me know if looks ok for testing